.circleci/config.yml

```diff
@@ -297,6 +298,244 @@ commands:
+              docker tag ${IMAGE}:${VERSION} ${GCP_IMAGE}:pr-${PR}-${SHORT_GIT_HASH}
+            fi
+
+            # tag images for release (based on tag)
+            if [[ ${CIRCLE_TAG} != "" ]];then
+              docker tag ${IMAGE}:${VERSION} taraxa/${IMAGE}:${CIRCLE_TAG}
+              docker tag ${IMAGE}:${VERSION} taraxa/${IMAGE}:latest
+            fi
+
+  push_gcr:
+    description: Push images to GCR registry
+    steps:
+      - run:
+          name: Login into GCR
+          command: |
+            echo ${GCLOUD_SERVICE_KEY} | docker login -u _json_key --password-stdin https://gcr.io
+
+      - run:
+          name: Push Images to GCR
+          command: |
+
+            # push images for 'develop' branch
+            if [[ ${CIRCLE_BRANCH} == "develop" ]];then
+              docker push ${GCP_IMAGE}-develop:${CIRCLE_SHA1}
+              docker push ${GCP_IMAGE}-develop:${SHORT_GIT_HASH}
+              docker push ${GCP_IMAGE}-develop:${DOCKER_BRANCH_TAG}-${CIRCLE_BUILD_NUM}
+              docker push ${GCP_IMAGE}-develop:${START_TIME}
+              docker push ${GCP_IMAGE}-develop:${START_TIME}-${CIRCLE_SHA1}
+              docker push ${GCP_IMAGE}-develop:latest
+            fi
+
+            # push images for 'release' branches
+            if [[ ${CIRCLE_BRANCH} =~ ^release.* ]]; then
+              docker push ${GCP_IMAGE}-release:${CIRCLE_SHA1}
+              docker push ${GCP_IMAGE}-release:${SHORT_GIT_HASH}
+              docker push ${GCP_IMAGE}-release:${DOCKER_BRANCH_TAG}-${CIRCLE_BUILD_NUM}
+              docker push ${GCP_IMAGE}-release:${START_TIME}
+              docker push ${GCP_IMAGE}-release:${START_TIME}-${CIRCLE_SHA1}
+              docker push ${GCP_IMAGE}-release:latest
+            fi
+
+            ### soon obsolete ###
+            if [[ ${CIRCLE_BRANCH} == "testnet" ]];then
+               echo ${DOCKERHUB_PASS} | docker login -u taraxa --password-stdin
+               docker push taraxa/${IMAGE}:${CIRCLE_SHA1}
+               docker push taraxa/${IMAGE}:${VERSION}
+               docker push taraxa/${IMAGE}:${START_TIME}
+               docker push taraxa/${IMAGE}:${START_TIME}-${CIRCLE_SHA1}
+               docker push taraxa/${IMAGE}:latest
+            fi
+            ### soon obsolete ###
+
+            # push images for 'feature' branches (based on PRs)
+            if [[ ${PR} != "false" ]]; then
+              docker push ${GCP_IMAGE}:pr-${PR}-${CIRCLE_BUILD_NUM}
+              docker push ${GCP_IMAGE}:pr-${PR}-${CIRCLE_SHA1}
+              docker push ${GCP_IMAGE}:pr-${PR}-${SHORT_GIT_HASH}
+            fi
+
+  push_dockerhub:
+    description: Push images to Docker Hub
+    steps:
+      - run:
+          name: Login into Docker Hub
+          command: |
+            echo ${DOCKERHUB_PASS} | docker login -u taraxa --password-stdin
+
+      - run:
+          name: Push Images
+          command: |
+            docker push taraxa/${IMAGE}:${CIRCLE_TAG}
```

Comment:
I think we should add the same conditions here. Only push if:
`if [[ ${CIRCLE_BRANCH} == "testnet" ]];then`
or
`if [[ ${CIRCLE_TAG} != "" ]];then`

Reply:
Nevermind. My bad.